.circleci/config.yml

```diff
@@ -0,0 +1,75 @@
+---
+version: 2.0
+
+references:
+  stack_build: &stack_build
+    docker:
+      # https://github.com/haskell-works/stack-build/blob/master/minimal/Dockerfile
+      - image: quay.io/haskell_works/stack-build-minimal
+
+    steps:
+      - checkout
+      - run:
+          name: Digest
+          command: |
+            echo -- "${STACK_ARGUMENTS}" > rdigest
+            git ls-files | xargs md5sum > sdigest
+
+      - restore_cache:
+          keys:
+            - 1-{{ .Branch }}-{{ checksum "rdigest" }}-{{ checksum "sdigest" }}
+            - 1-{{ .Branch }}-{{ checksum "rdigest" }}-
+            - 1-{{ .Branch }}-
+            - 1-master-
+
+      - run:
+          name: Dependencies
+          command: |
+            stack setup $STACK_ARGUMENTS
+            stack build stratosphere $STACK_ARGUMENTS --dependencies-only \
+            --test --no-run-tests --jobs=1 --flag stratosphere:-library-only
+          no_output_timeout: 30m
+
+      - run:
+          name: Build
+          command: |
+            stack build stratosphere $STACK_ARGUMENTS --pedantic --test \
+              --no-run-tests --jobs=1 --flag stratosphere:-library-only
+          no_output_timeout: 30m
+
+      - save_cache:
+          key: 1-{{ .Branch }}-{{ checksum "rdigest" }}-{{ checksum "sdigest" }}
+          paths:
+            - ~/.stack
+            - ./.stack-work
+
+      - run:
+          name: Test
+          command: |
+            stack build stratosphere $STACK_ARGUMENTS --pedantic --test \
+            --jobs=1 --flag stratosphere:-library-only
+
+      - run:
+          name: Docs
```

Comment:
Ugh, Haddock unregisters and rebuilds? That sucks hard haha. Maybe I'll split out a special job just for building docs. We don't need to build docs on the old GHC anyway. (You don't have to do this if you don't want to, I can do it).

Reply:
I should be able to do it!